### PR TITLE
List structure

### PR DESCRIPTION
**Pull request by gisachris**
## IN This PR I present the To-Do-List app structure :
Below are some of my implementations as well as going over the project requirements 🤝 .
## Requirements
-build a basic page structure for the to-do-list app 🧑‍🔧 
-build the structure using Webpack 🟦 
-add the necessary logic 🧠 to add items to the list and to the Dom  📜 
-add styling 🎨  based off the list structure given 

## Achievements
-build a basic page structure 🥼  for the to-do-list app 🎖️ 
-build the structure using Webpack 🟦 
-add items to their holder and display them on the Dom  📜
-add the necessary styling 🎨

## N.B
**things like local storage and full page functionality where not a requirement at this point of the project and I why there not implemented in this pull request**

## Conclusion
**I would love to hear 👂  from you my CR in terms of what can be better implemented 👨‍💻  for the site thankyou!. And I look forward to hearing from you soon.**  🙇

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,14 +1,107 @@
...
+arrowHolder.src = arrow;
+
+inputSection.append(inputField, arrowHolder);
+
+// selecting the button
+const button = document.querySelector('.clearAll');
+
+// implementing list functionality
+
+const taskHolder = [];
```
Comment (bobb-Rob) [inline src/index.js]: - [ ] Based on the project requirement, it is expected that on page load, you render dynamically, the created list of tasks in the dedicated placeholder. The list should appear in order of the index values for each task. I suggest you use Windows `load` events which fire when the **_whole page_** has loaded, including all dependent resources such as stylesheets, scripts, iframes, and images.

```javascript
  window.addEventListener("load", renderTasks);
```

The `renderTasks` function in the code snippet above is the same as your `addToDom` function. 

Kindly implement this.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @gisachris, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gisachris** [comment]
## After my previous CodeReview I was asked to implement some changes.
## Changes requested
**-Add an event listener 👂  for Windows load to dynamically ⛓️  load list section script 📜  contents.**
## Implementations
**-Add the event listener to load list section dynamically 💹**
## Conclusion
**I would love to here from you my CR on any further changes I can implement for this stage of the project 🙇**

File: src/index.js [hunk 2]
```diff
@@ -1,14 +1,110 @@
...
```
Comment (braincee) [inline src/index.js]: - [ ] As requested by the previous reviewer, it is expected that you render a created list of tasks dynamically per the [project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m1_list_structure.md). I highly suggest you populate your `const taskHolder = [];` which is an empty array with data.
This is an example of how to populate the empty array with data:

```
const taskHolder = [
  { description: 'task1', completed: false, index: 1 },
    { description: 'task2', completed: true, index: 2 },
    { description: 'task3', completed: false, index: 3 },
];

```

This is a screenshot of how the populated data should render dynamically. 😄 

Screenshot (296)

This is a screenshot of your project. 😄 

Screenshot (298)

**Review by braincee — CHANGES_REQUESTED** [review]
Hello @gisachris 👋 , I am @braincee your Code Reviewer

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


so-close

HIGHLIGHTS
------

 - GitHub flow rightfully used ✅
 - Meaningful title and summary for PR ✅
 - No linter errors ✅

REQUIRED CHANGES ♻️
------
**_Please check the comments under the review._**

OPTIONAL CHANGES
--------
**_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @braincee in your question so I can receive the notification.

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

----

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (gisachris) [reply src/index.js]: ok thankyou the previous reviewer didn't clearly specify what needed to be done 👍

Comment (braincee) [reply src/index.js]: > ok thankyou the previous reviewer didn't clearly specify what needed to be done 👍

You welcome

**Comment by gisachris** [comment]
## After my previous CodeReview I was asked to implement some changes.
## Changes requested
**-implement a static array of objects and load them as a list dynamically ⛓️ in the script 📜  using windows load event listener 👂 .**
## Implementations
**-Add the event listener to load the list section dynamically 💹**
## Conclusion
**My first codeReviwer was not very specific on what had to be changed But the last reviewer made the changes clear and I have implemented them .I would love to hear from you my CR on any further changes I can implement for this stage of the project 🙇**

**Review by braincee — APPROVED** [review]
Hi @gisachris 👋, I am @braincee, your Code Reviewer.

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

youre-approved-shirley-auto-group

OPTIONAL CHANGES
---

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @braincee._

---

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**